### PR TITLE
Updated the PPS PCL workers to support the new AlCaRecoProducer

### PR DESCRIPTION
**Pull request by MatiXOfficial**
#### PR description:

This PR modifies the way the PCL workers (alignment, timing, sampic timing) handle consuming input products. Single InputTags have been replaced with vectors of InputTags so that the workers could work both with the old producer and with the new producer introduced in https://github.com/cms-sw/cmssw/pull/36273.

By default, the behaviour of the workers is unchanged - the vectors contain only one element. The PCL configs modify these vectors to contain the InputTags required by the new producer as well.

In the case of the timing PCLs, the re-reconstruction of the recHits was dropped, and the old producer was modified to supply the recHits directly. It shouldn't cause any change in terms of the produced results.

#### PR validation:

- no changes in the relvals (1041, 1042)
- no changes in the results of some local tests
- The multi-InputTag policy works as expected - the PCL workers accept the files produced by the old producer and the files produced by the new producer.

### CODE REVIEW

**Comment by tvami** [comment]
test parameters:
  - workflows = 1041, 1042

File: CalibPPS/AlignmentGlobal/plugins/PPSAlignmentWorker.cc [hunk 1]
```diff
@@ -91,7 +91,8 @@ class PPSAlignmentWorker : public DQMEDAnalyzer {
   edm::ESGetToken<PPSAlignmentConfiguration, PPSAlignmentConfigurationRcd> esTokenBookHistograms_;
   edm::ESGetToken<PPSAlignmentConfiguration, PPSAlignmentConfigurationRcd> esTokenAnalyze_;
 
-  edm::EDGetTokenT<CTPPSLocalTrackLiteCollection> tracksToken_;
+  std::vector<edm::InputTag> tracksTags_;
+  std::vector<edm::EDGetTokenT<CTPPSLocalTrackLiteCollection>> tracksTokens_;
```
Comment (tvami) [inline CalibPPS/AlignmentGlobal/plugins/PPSAlignmentWorker.cc:95]: can these be `const`?

File: CalibPPS/TimingCalibration/plugins/PPSDiamondSampicTimingCalibrationPCLWorker.cc [hunk 2]
```diff
@@ -62,22 +62,35 @@ class PPSDiamondSampicTimingCalibrationPCLWorker
...
+                        edm::Handle<T>& handle) const;
+
   // ------------ member data ------------
-  edm::EDGetTokenT<edm::DetSetVector<TotemTimingDigi>> totemTimingDigiToken_;
-  edm::EDGetTokenT<edm::DetSetVector<TotemTimingRecHit>> totemTimingRecHitToken_;
+  std::vector<edm::InputTag> digiTags_;
+  std::vector<edm::InputTag> RecHitTags_;
+
+  std::vector<edm::EDGetTokenT<edm::DetSetVector<TotemTimingDigi>>> totemTimingDigiTokens_;
+  std::vector<edm::EDGetTokenT<edm::DetSetVector<TotemTimingRecHit>>> totemTimingRecHitTokens_;
```
Comment (tvami) [inline CalibPPS/TimingCalibration/plugins/PPSDiamondSampicTimingCalibrationPCLWorker.cc:76]: similarly can these be `const`-s?

File: CalibPPS/TimingCalibration/plugins/PPSTimingCalibrationPCLWorker.cc [hunk 3]
```diff
@@ -41,7 +41,14 @@ class PPSTimingCalibrationPCLWorker : public DQMGlobalEDAnalyzer<TimingCalibrati
...
-  edm::EDGetTokenT<edm::DetSetVector<CTPPSDiamondRecHit>> diamondRecHitToken_;
+  template <typename T>
+  bool searchForProduct(edm::Event const& iEvent,
+                        const std::vector<edm::EDGetTokenT<T>>& tokens,
+                        const std::vector<edm::InputTag>& tags,
+                        edm::Handle<T>& handle) const;
+
+  std::vector<edm::InputTag> RecHitTags_;
+  std::vector<edm::EDGetTokenT<edm::DetSetVector<CTPPSDiamondRecHit>>> diamondRecHitTokens_;
   edm::ESGetToken<CTPPSGeometry, VeryForwardRealGeometryRecord> geomEsToken_;
```
Comment (tvami) [inline CalibPPS/TimingCalibration/plugins/PPSTimingCalibrationPCLWorker.cc:52]: I think the ESGetToken for sure can be const :)

File: CalibPPS/TimingCalibration/plugins/PPSTimingCalibrationPCLWorker.cc [hunk 4]
```diff
@@ -85,9 +94,10 @@ void PPSTimingCalibrationPCLWorker::bookHistograms(DQMStore::IBooker& iBooker,
 void PPSTimingCalibrationPCLWorker::dqmAnalyze(const edm::Event& iEvent,
                                                const edm::EventSetup& iSetup,
                                                const TimingCalibrationHistograms& iHists) const {
-  // then extract the rechits information for later processing
   edm::Handle<edm::DetSetVector<CTPPSDiamondRecHit>> dsv_rechits;
```
Comment (tvami) [inline CalibPPS/TimingCalibration/plugins/PPSTimingCalibrationPCLWorker.cc:97]: Maybe const here too?

File: CalibPPS/AlignmentGlobal/plugins/PPSAlignmentWorker.cc (hunk 1 above)
Comment (MatiXOfficial) [reply CalibPPS/AlignmentGlobal/plugins/PPSAlignmentWorker.cc:95]: Good idea, but only `tracksTags_` can be const. `tracksTokens_` cannot be initialised in the initializer list.

File: CalibPPS/TimingCalibration/plugins/PPSTimingCalibrationPCLWorker.cc (hunk 3 above)
Comment (MatiXOfficial) [reply CalibPPS/TimingCalibration/plugins/PPSTimingCalibrationPCLWorker.cc:52]: Yes :) and the `tags` variable too

File: CalibPPS/TimingCalibration/plugins/PPSTimingCalibrationPCLWorker.cc (hunk 4 above)
Comment (MatiXOfficial) [reply CalibPPS/TimingCalibration/plugins/PPSTimingCalibrationPCLWorker.cc:97]: I don't think `dsv_rechits` can be const. It is assigned in `searchForProduct`.

File: CalibPPS/TimingCalibration/plugins/PPSDiamondSampicTimingCalibrationPCLWorker.cc (hunk 2 above)
Comment (MatiXOfficial) [reply CalibPPS/TimingCalibration/plugins/PPSDiamondSampicTimingCalibrationPCLWorker.cc:76]: Ok, to sum up, I added const to every member variable in these workers that can be const.

**Comment by tvami** [comment]
@cmsbuild , please test

File: CalibPPS/TimingCalibration/plugins/PPSDiamondSampicTimingCalibrationPCLWorker.cc (hunk 2 above)
Comment (tvami) [reply CalibPPS/TimingCalibration/plugins/PPSDiamondSampicTimingCalibrationPCLWorker.cc:76]: Thanks @MatiXOfficial !

**Comment by tvami** [comment]
@cmsbuild , please test

**Comment by tvami** [comment]
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-975475/22627/addOnTests/hlt_mc_PRef/cmsDriver.py_RelVal_-s_HLT:PRef,RAW2DIGI,L1Reco,RECO_--mc_--scenario=pp_-n_10_--conditions_auto:run3_mc_PRef_--relval_9000,50_--datatier_RAW-HLT-RECO.log

This doesnt seem related...

**Comment by tvami** [comment]
@cmsbuild , please test

**Comment by tvami** [comment]
+alca
 * The output of 1041 and 1042 looks good
 * no diffs in Jenkins

**Comment by qliphy** [comment]
+1

**Pull request merged**